File: vscode-extension/src/protocol/NodeData.ts
/*------------------------------------------------------------------------------------------------
 *  Copyright (c) 2022 ICSharpCode
 *  Licensed under the MIT License. See LICENSE.TXT in the project root for license information.
 *-----------------------------------------------------------------------------------------------*/

import Node from "./Node";

export default interface NodeData {
  node?: Node;
  name: string;
  description: string;
  mayHaveChildren: boolean;
}

File: backend/src/ILSpy.Backend/Model/NodeData.cs
// Copyright (c) 2022 ICSharpCode
// Licensed under the MIT license. See the LICENSE file in the project

namespace ILSpy.Backend.Model;

public record NodeData(
    Node? Node,
    string DisplayName,
    string Description,
    bool MayHaveChildren,
    SymbolModifiers SymbolModifiers);
